source/geometry_gmsh.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "geometry_common_includes.h"
+
+#include <deal.II/grid/grid_in.h>
+
+#include <fstream>
+
+namespace ryujin
+{
+  namespace Geometries
+  {
+    /**
+     * This class imports a triangulation from a GMSH *.msh file.
+     *
+     * @ingroup Mesh
+     */
+    template <int dim>
+    class GMSH : public Geometry<dim>
+    {
+    public:
+      GMSH(const std::string subsection)
+          : Geometry<dim>("gmsh", subsection)
+      {
+        filename_ = "ryujin.msh";
+        this->add_parameter("filename", filename_, "GMSH *.msh file to load");
+      }
+
+      void create_triangulation(
+          typename Geometry<dim>::Triangulation &triangulation) final
+      {
+        dealii::GridIn<dim> gridin;
+        gridin.attach_triangulation(triangulation);
+        std::ifstream file(filename_);
+        gridin.read_msh(file);
```

Comment:
The logic is so small that we can simply do the call to `GridIn` in the `create_triangulation()` function.

(The split that you copied where the implementation is split out into a `GridGenerator` function served a special purpose to be closer to the deal.II `GridGenerator` functions. But that isn't needed here.)